File: LinkedLists/Singly-LinkedList/LinkedList.py
#Establishing Node
class Node:

    def __init__(self):
        self.data = None
        self.Next = None

#Establishing List
class LinkedList:

    #Initializing List
    def __init__(self):
        self.head = None
        self.size = 0


    #Check whether the list is Empty or not
    def isEmpty(self):
        return self.head == None


    #Traversing The List
    def traverse(self):

        if(self.isEmpty()):
            print("List is Empty!")
            return

        currentNode = self.head
        
        while(currentNode != None):
            print(currentNode.data, end = " -> ")
            currentNode = currentNode.Next;

        print()


    #Inserting Node(s)
    def insert(self, x, loc):

        newNode = Node()
        newNode.data = x

        if loc == 0:

            newNode.Next = self.head
            self.head = newNode;

        else:

            if loc > self.size:

                print("The list is only {} nodes".format(self.size))
                return

            currentNode = self.head

            for i in range(loc - 1):
                currentNode = currentNode.Next

            newNode.Next = currentNode.Next
            currentNode.Next = newNode

        self.size += 1


    #Deleting Node(s)
    def delete(self, loc):

        if(self.isEmpty()):

            print("List is Already Empty!")
            return

        if(loc > self.size - 1):
            print("The list is only {} nodes!".format(self.size))
            return

        currentNode = self.head
        previousNode = self.head

        if loc == 0:

            self.head = currentNode.Next

        elif loc == self.size - 1:
            
            for i in range(loc - 1):
                currentNode = currentNode.Next

            currentNode.Next = None
            
        else:

            for i in range(loc):

                currentNode = currentNode.Next

                if i == 0:
                    continue

                previousNode = previousNode.Next

            previousNode.Next = currentNode.Next

        self.size -= 1

File: LinkedLists/Singly-LinkedList/LinkedList.java
public class LinkedList {

    //Establishing Node
    class Node {

        int data;
        Node next;

    };

    private Node head;
    private int size;

    //Initializing The List
    LinkedList(){

        head = null;
        size = 0;

    }

    //Check whether the list is Empty or not
    public boolean isEmpty(){
        return head == null;
    }

    //Traversing The List
    public void traverse(){

        if(isEmpty()){
            System.out.println("List is Empty!");
            return;
        }

        Node currentNode = head;

        while (currentNode != null){
            System.out.print(currentNode.data + " -> ");
            currentNode = currentNode.next;
        }

        System.out.println();

    }

    //Inserting Node(s)
    public void insert(int x, int loc){

        Node newNode = new Node();
        newNode.data = x;

        if(loc == 0){

            newNode.next = head;
            head = newNode;

        } else {

            if(loc > size){
                System.out.println("The list is only " + size + " nodes!");
                return;
            }

            Node currentNode = head;

            for(int i = 0; i < loc - 1; i++)
                currentNode = currentNode.next;

            newNode.next = currentNode.next;
            currentNode.next = newNode;

        }

        size++;
    }

    //Deleting Node(s)
    public void delete(int loc){

        if(isEmpty()){
            System.out.println("List is already Empty!");
            return;
        }

        if(loc > size - 1){
            System.out.println("The list is only " + size + " node");
            return;
        }

        Node currentNode = head;
        Node previousNode = head;

        if(loc == 0){

            head = currentNode.next;

        } else if (loc == size - 1){

            for(int i = 0; i < loc - 1; i++)
                currentNode = currentNode.next;

            currentNode.next = null;

        } else {

            for(int i = 0; i < loc; i++) {
                currentNode = currentNode.next;

                if(i == 0)
                    continue;

                previousNode = previousNode.next;

            }

            previousNode.next = currentNode.next;

        }

        size--;

    }



}
